### PR TITLE
feat: shorten the value of the contact field

### PR DESCRIPTION
**Pull request by asbiin**
This comes after #2288, which propose to use the name of the contact field instead of the value.
I propose to shorten the value if it's too long (26 characters seems good), and add a '...' after.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -18,7 +18,7 @@ Enhancements:
 * Register all app services as singleton
 * Docker image: add sentry-cli and run sentry:release command if sentry is enabled
 * Refactor reminders by removing Notifications table and creating two new tables: reminder outbox and reminder sent
-* Display the name of the contact field instead of the URL (Facebook instead of https:/facebook.com/johndoe)
+* Shorten the value of the contact field if it not fits into the contact field information
```
Comment (djaiss) [inline CHANGELOG]:
```suggestion
* Shorten the value of the contact field if it does not fit into the contact field information
```

File: resources/assets/js/components/people/ContactInformation.vue [hunk 2]
```diff
@@ -33,10 +33,10 @@
             <i v-else class="pr2 fa fa-address-card-o f6 gray"></i>
 
             <a v-if="contactInformation.protocol" :href="contactInformation.protocol + contactInformation.data">
-              {{ contactInformation.name }}
+              {{ contactInformation.short }}
```
Comment (djaiss) [inline resources/assets/js/components/people/ContactInformation.vue]:
```suggestion
              {{ contactInformation.shortenName }}
```

File: resources/assets/js/components/people/ContactInformation.vue [hunk 3]
```diff
@@ -33,10 +33,10 @@
...
             </a>
             <a v-else :href="contactInformation.data">
-              {{ contactInformation.name }}
+              {{ contactInformation.short }}
```
Comment (djaiss) [inline resources/assets/js/components/people/ContactInformation.vue]:
```suggestion
              {{ contactInformation.shortenName }}
```

File: resources/assets/js/components/people/ContactInformation.vue [hunk 4]
```diff
@@ -165,10 +169,22 @@ export default {
...
         .then(response => {
-          this.contactInformationData = response.data;
+          this.contactInformationData = this.formatResponse(response.data);
         });
     },
 
+    formatResponse(data) {
+      var vm = this;
+      _.each(data, function(value) {
+        var short = value.data;
```
Comment (djaiss) [inline resources/assets/js/components/people/ContactInformation.vue]:
```suggestion
        var shortenName = value.data;
```

File: resources/assets/js/components/people/ContactInformation.vue [hunk 5]
```diff
@@ -165,10 +169,22 @@ export default {
...
+        if (short.length > vm.sizeLimit + 1) {
+          short = vm.$t('format.short_text', { text: short.substr(0, vm.sizeLimit) });
```
Comment (djaiss) [inline resources/assets/js/components/people/ContactInformation.vue]:
```suggestion
          shortenName = vm.$t('format.short_text', { text: shortenName.substr(0, vm.sizeLimit) });
```

File: resources/assets/js/components/people/ContactInformation.vue [hunk 6]
```diff
@@ -165,10 +169,22 @@ export default {
...
+        }
+        value.short = short;
```
Comment (djaiss) [inline resources/assets/js/components/people/ContactInformation.vue]:
```suggestion
        value.shortenName = shortenName;
```

**Comment by asbiin** [comment]
@djaiss thanks for the review

**Pull request merged**